services/filemanager/azureBlobStoragemanager.go

```diff
@@ -105,10 +123,26 @@ func (manager *AzureBlobStorageManager) Upload(ctx context.Context, file *os.Fil
 		return UploadOutput{}, err
 	}
 
-	return UploadOutput{Location: blobURL.String(), ObjectName: fileName}, nil
+	return UploadOutput{Location: manager.blobLocation(&blobURL), ObjectName: fileName}, nil
 }
 
-func (manager *AzureBlobStorageManager) ListFilesWithPrefix(ctx context.Context, startAfter, prefix string, maxItems int64) (fileObjects []*FileObject, err error) {
+func (manager *AzureBlobStorageManager) shouldCreateContainer() bool {
+	return !manager.useSASTokens()
+}
+
+func (manager *AzureBlobStorageManager) blobLocation(blobURL *azblob.BlockBlobURL) string {
+	if !manager.useSASTokens() {
+		return blobURL.String()
+	}
+
+	// Reset SAS Query parameters
+	blobURLParts := azblob.NewBlobURLParts(blobURL.URL())
+	blobURLParts.SAS = azblob.SASQueryParameters{}
+	newBlobURL := blobURLParts.URL()
+	return newBlobURL.String()
```

Comment:
minor and subjective
```suggestion
	return blobURLParts.URL().String()
```

Reply:
Will make the changes.

Reply:
It's giving this error `Cannot call a pointer method on 'blobURLParts.URL()'`

Reply:
Leaving it as it is.